lib/utensils/have_hash_matcher.rb

```diff
@@ -0,0 +1,60 @@
+RSpec::Matchers.define :have_hash do |expected|
+  match do |array|
+    synchronize do
+      hash = find_hash(array, expected)
+
+      expect(hash).to include(expected)
+      expect(hash).to_not include(@unexpected_keys)
+    end
+  end
+
+  chain :without do |*unexpected_keys|
+    @unexpected_keys = unexpected_keys
+  end
+
+  failure_message do
+    @failure_message
+  end
+
+  private
+
+    def find_hash(array, expected)
+      hash = array.reverse.find do |item|
+        find_with_inclusion(item, expected)
+      end
+
+      hash || better_diff(array)
+    end
+
+    def find_with_inclusion(hash, expected)
+      expect(hash).to include(expected)
+    rescue RSpec::Expectations::ExpectationNotMetError
+    end
+
+    def better_diff(array)
+      return array if array.size > 1
+
+      array.first
+    end
+
+    # mimicking has_text/css? matchers, https://git.io/vMSo9
+    def synchronize(&block)
+      start_time = Time.current
+
+      begin
+        block.call
+      rescue ::RSpec::Expectations::ExpectationNotMetError => e
+        @failure_message = e.message
+        raise(e) unless js_enabled?
+
+        return false if (Time.current - start_time) >= Capybara.default_max_wait_time
+
+        sleep 0.05
+        retry
+      end
+    end
+
+    def js_enabled?
+      RSpec.current_example.metadata[:js]
```

Comment:
I implemented a version of [`js_enabled?`](https://github.com/cookpad/global-web/blob/7daaafd0cccba531ed7388a8a730a68b4da50477/spec/support/helpers/js_driver_helper.rb#L3) that checks the capabilities of the browser rather than relying on metadata tags. This approach seemed more reliable and stable if the metadata tags changed.

```javascript
  def js_enabled?
    !!page.evaluate_script("0 + 1")
  rescue Capybara::NotSupportedByDriverError
    false
  end
```

And along those lines, I think our metadata tags aren't the most logical and would like to refactor them if I get some time.